### DISCUSSION TITLE
register_message_handler question

### DISCUSSION THREAD

**Discussion by kelogsloops** [eternnoir/pyTelegramBotAPI#2005] [Q&A]
Please answer these questions before submitting your issue. Thanks!

1. What version of pyTelegramBotAPI are you using?
latest
2. What OS are you using?
debian
3. What version of python are you using?
3.7

In a normal bot, I use the send_photo method to send a photo to the user with a set of reply keystrokes, called message_a. Afterwards I use the register_message_handler method to monitor bot.register_message_handler( callback = self.start_panel, content_types = ['text'],regexp = "Button_A"). I want to delete message_a when the user clicks on Button_A, but the only parameter passed to register_message_handler is the message for "Button_A". Is it not possible to pass the message of "message_A" into the callback function as well? How can I improve this?

**Comment by coder2020official** [comment]
I don't get you. Can you explain what are you trying to achieve in your bot?

**Comment by kelogsloops** [comment]
> I don't get you. Can you explain what are you trying to achieve in your bot?

bot sends messageA with a reply to the keyboards "buttonA" and "buttonB". I want the user to delete messageA and send a new messageB when they click on "buttonA".

**Comment by coder2020official** [comment]
Ok, so you are sending a message with 2 buttons and you need to delete the message when one of the buttons is clicked?

**Comment by kelogsloops** [comment]
> Ok, so you are sending a message with 2 buttons and you need to delete the message when one of the buttons is clicked?

Yes, that's right.

**Comment by coder2020official** [comment]
And what's the issue?
Create a callback query handler to catch events of buttons getting clicked and use bot.delete_message and send a message

**Comment by kelogsloops** [comment]
> And what's the issue? Create a callback query handler to catch events of buttons getting clicked and use bot.delete_message and send a message

I use register_message_handler to monitor the button, but the only callback I pass in is the button's message, not the message I want to delete.

**Comment by kelogsloops** [comment]
The build button utilises not InlineKeyboardMarkup, but ReplyKeyboardMarkup

**Comment by coder2020official** [comment]
Maybe you are using register_next_step_handler and not message_handler?

**Comment by kelogsloops** [comment]
> Maybe you are using register_next_step_handler and not message_handler?

I am trying register_next_step_handler, but can't message_handler be implemented in my scenario?
